### DISCUSSION TITLE
Hide a page from the sitemap

### DISCUSSION THREAD

**Discussion by Le0Developer** [withastro/starlight#2100] [Feature Requests]
### What version of `starlight` are you using?

0.17.2

### What is your idea?

Add a `hidden` tag or similar to frontmatter that makes the page not appear on the sitemap (and/or search results?).

### Why is this feature necessary?

We need to exclude certain (in dev or private) documents from the sitemap to prevent accidental indexing by search engines before they are ready.

### Participation

- [ ] I am willing to submit a pull request for this feature.

**Comment by delucis** [comment]
Have you considered using the [`draft` frontmatter option](https://starlight.astro.build/reference/frontmatter/#draft)? This was added in Starlight v0.22, so you would need to upgrade to use it.

**Reply by Le0Developer** [reply]
We still want our in-dev pages to be included in production builds, so they're easily accessible for review.
We also have some private docs that are password protected in our reverse proxy, so them not being in the sitemap would prevent search engines from crawling them (and failing).

**Reply by delucis** [reply]
Thanks for the extra feedback.

One other workaround that is possible today is to add [`@astrojs/sitemap`](https://docs.astro.build/en/guides/integrations-guide/sitemap/) manually and configure it yourself — this will then replace Starlight’s internal implementation of the sitemap.
